java/arcs/core/data/Plan.kt

```diff
@@ -37,6 +37,7 @@ open class Plan(
     /** Represents a use of a [Handle] by a [Particle]. */
     data class HandleConnection(
         var storageKey: StorageKey,
+        var mode: HandleMode,
```

Comment:
Does it need to be `var` or can it be `val`?

Reply:
`val` would be ideal

Reply:
Kind of curious why `storageKey` is `var` too actually.

Reply:
it should be val, good catch. StorageKey is var because create keys are replaced-in-place by the Allocator, but this could be changed to copy() construct a new HandleConnection, something for post march, or for Shane to clean up as he is implementing.